user-impl/src/main/java/com/example/auction/user/impl/UserRepository.java

```diff
@@ -0,0 +1,181 @@
+                                    CompletableFuture.completedFuture(TreePVector.empty()) :
+                                    selectUsers(offset, limit);
+                            return Users.thenApply(seq -> new PaginatedSequence<>(seq, page, pageSize, count));
+                        }
+                );
+    }
+
+    private CompletionStage<Integer> countUsers() {
+        return session
+                .selectOne(
+                        "SELECT COUNT(*) FROM UserInfo ORDER BY UserId DESC"
+                )
+                .thenApply(row -> (int) row.get().getLong("count"));
+    }
+
+    private CompletionStage<PSequence<User>> selectUsers(long offset, int limit) {
+
+        return session
+                .selectAll(
+                        "SELECT * FROM UserInfo " +
+                                "ORDER BY  UserId DESC " +
+                                "LIMIT ?",
+                        limit
+                )
+                .thenApply(List::stream)
+                .thenApply(rows -> rows.skip(offset))
+                .thenApply(rows -> rows.map(UserRepository::convertUserSummary))
+                .thenApply(UserSummaries -> UserSummaries.collect(Collectors.toList()))
+                .thenApply(TreePVector::from);
+    }
+
+
+    private static User convertUserSummary(Row user) {
+        return new User(
+                user.getUUID("UserId"),
+                user.getString("Name"),
+                user.getString("email")
+        );
+    }
+
+    private static class PUserEventProcessor extends ReadSideProcessor<PUserEvent> {
+
+        private final CassandraSession session;
+        private final CassandraReadSide readSide;
+
+        private PreparedStatement insertUserStatement;
+
+        @Inject
+        public PUserEventProcessor(CassandraSession session, CassandraReadSide readSide) {
+            this.session = session;
+            this.readSide = readSide;
+        }
+
+        @Override
+        public ReadSideHandler<PUserEvent> buildHandler() {
+            return readSide.<PUserEvent>builder("pUserEventOffset")
+                    .setGlobalPrepare(this::createTables)
+                    .setPrepare(tag -> prepareStatements())
+                    .setEventHandler(PUserEvent.PUserCreated.class,
+                            e -> insertUser(e.getUser()))
+
+                    .build();
+        }
+
+        @Override
+        public PSequence<AggregateEventTag<PUserEvent>> aggregateTags() {
+            return PUserEvent.TAG.allTags();
+        }
+
+        private CompletionStage<Done> createTables() {
+            return doAll(
+                    session.executeCreateTable(
+                            "CREATE TABLE IF NOT EXISTS UserInfo (" +
+                                    "UserId UUID PRIMARY KEY, " +
+                                    "Name text, " +
+                                    "email text, " +
```

Comment:
This is not a syntax error, on the contrary.

If I remove the trailing comma I get a syntax error from the cassandra driver.
`ItemRepository` also have a comma separating a the last column and the definition of a compound primary key.

There is two slightly different syntax.
```
CREATE TABLE demo (
 foo text PRIMARY KEY,
 bar text
)
```
No comma between `foo` column definition and `PRIMARY KEY` as we are marking `foo` as the primary key.

```
CREATE TABLE demo (
 foo text,
 bar text,
 PRIMARY KEY(foo, bar)
)
```
In the above example we do need a comma because we are defining a compound PK. In that case, "PRIMARY KEY" is not part of the column definition and therefore a comma is needed